src/coreclr/jit/importercalls.cpp

```diff
@@ -104,13 +104,17 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
     bool checkForSmallType  = false;
     bool bIntrinsicImported = false;
 
-    CORINFO_SIG_INFO calliSig;
+    CORINFO_SIG_INFO originalSig = {};
     NewCallArg       extraArg;
 
-    /*-------------------------------------------------------------------------
-     * First create the call node
-     */
+    // run transformations when instrumenting to not pollute PGO data
+    bool                  optimizedOrInstrumented = opts.OptimizationEnabled() || opts.IsInstrumented();
+    CORINFO_METHOD_HANDLE replacementMethod       = nullptr;
+    GenTree*              newThis                 = nullptr;
+    var_types             oldThis                 = TYP_UNDEF;
+    var_types             targetThis              = TYP_UNDEF;
```

Comment:
I would rename these to be a bit more descriptive, eg `oldThisTyp`.

Also move the declaration/definition down closer to where they're used.